flang/include/flang/Tools/CLOptions.inc

```diff
@@ -70,40 +70,16 @@ inline void addCSE(mlir::PassManager &pm) {
       pm, disableFirCse, fir::createCSEPass);
 }
 
-#if !defined(FLANG_EXCLUDE_CODEGEN)
-inline void addCodeGenRewritePass(mlir::PassManager &pm) {
-  addPassConditionally(
-      pm, disableCodeGenRewrite, fir::createFirCodeGenRewritePass);
-}
-
-inline void addTargetRewritePass(mlir::PassManager &pm) {
-  addPassConditionally(pm, disableTargetRewrite, []() {
-    return fir::createFirTargetRewritePass(fir::TargetRewriteOptions{});
-  });
-}
-
-inline void addFIRToLLVMPass(mlir::PassManager &pm) {
-  addPassConditionally(pm, disableFirToLlvmIr, fir::createFIRToLLVMPass);
-}
-
-inline void addLLVMDialectToLLVMPass(
-    mlir::PassManager &pm, llvm::raw_ostream &output) {
-  addPassConditionally(pm, disableLlvmIrToLlvm,
-      [&]() { return fir::createLLVMDialectToLLVMPass(output); });
-}
-
-/// Create a pass pipeline for lowering from MLIR to LLVM IR
+/// Create a basic pass pipeline for simplyfing the FIR dialect
 ///
 /// \param pm - MLIR pass manager that will hold the pipeline definition
-inline void createMLIRToLLVMPassPipeline(mlir::PassManager &pm)
+inline void createBasicFirPassPipeline(mlir::PassManager &pm)
 {
   // simplify the IR
-  mlir::GreedyRewriteConfig config;
-  config.enableRegionSimplification = false;
   fir::addCSE(pm);
   pm.addNestedPass<mlir::FuncOp>(fir::createArrayValueCopyPass());
   pm.addNestedPass<mlir::FuncOp>(fir::createCharacterConversionPass());
-  pm.addPass(mlir::createCanonicalizerPass(config));
+  pm.addPass(mlir::createCanonicalizerPass());
```

Comment:
This looks incorrect. Running MLIR's full canonicalizer pass deploys features that break the compiler.

Reply:
This was copied from `bbc` (so that the code can be shared between `bbc` and `tco`/`flang-new`). Is `bbc` incorrect then? (extracted from [here](https://github.com/flang-compiler/f18-llvm-project/blob/fir-dev/flang/tools/bbc/bbc.cpp#L270-L294)). Or are `bbc` and `tco`/`flang-new` meant to diverge here?